pkg/reconcilers/base_reconciler.go

```diff
@@ -31,6 +31,44 @@ func CreateOnlyMutator(existing, desired common.KubernetesObject) (bool, error)
 	return false, nil
 }
 
+// Remove 'ThreeScaleApicastRequestTime' alert
+func RemovePrometheusRulesMutator(existing, desired common.KubernetesObject) (bool, error) {
+	existingPrometheusRule := existing.(*monitoringv1.PrometheusRule)
```

Comment:
The implementation is correct. HAve you considered using https://pkg.go.dev/golang.org/x/exp/slices ?